compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1479,6 +1565,14 @@ public void visit(BLangAssignment assignNode) {
     public void visit(BLangTupleDestructure tupleDeStmt) {
         for (BLangExpression tupleVar : tupleDeStmt.varRef.expressions) {
             setTypeOfVarRefForBindingPattern(tupleVar);
+            if (tupleVar.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                BLangSimpleVarRef variableRef = (BLangSimpleVarRef) tupleVar;
+                if (variableRef.lhsVar && variableRef.symbol != null &&
+                        (variableRef.symbol.tag & SymTag.TYPE_DEF) == SymTag.TYPE_DEF) {
+                    dlog.error(tupleVar.pos, DiagnosticErrorCode.CANNOT_ASSIGN_VALUE_TO_TYPE_DEF);
+                    return;
```

Comment:
Why are we returning here? Can't we continue the rest of the analysis? Same with other changes too.